oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/query/async/facets/ElasticSecureFacetAsyncProvider.java

```diff
@@ -112,16 +111,20 @@ public void endData() {
                                 .collect(Collectors.toList())
                         )
                 );
-        LOG.trace("End data {}", facetsMap);
+        LOG.trace("End data {}", facets);
         latch.countDown();
     }
 
     @Override
     public List<FulltextIndex.Facet> getFacets(int numberOfFacets, String columnName) {
         LOG.trace("Requested facets for {} - Latch count: {}", columnName, latch.getCount());
         try {
-            latch.await(15, TimeUnit.SECONDS);
+            boolean completed = latch.await(15, TimeUnit.SECONDS);
```

Comment:
make configurable? If the facets query is very slow, we may want to increase the timeout. Could it ever take longer than 15 seconds?

Reply:
Makes sense. But this should stay in a separate PR.